qiskit/algorithms/linear_solvers/matrices/linear_system_matrix.py

```diff
@@ -0,0 +1,137 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""An abstract class for linear systems solvers in Qiskit's aqua module."""
+
+from abc import ABC, abstractmethod
+
+from qiskit import QuantumCircuit
+from qiskit.circuit.library import BlueprintCircuit
+
+
+class LinearSystemMatrix(BlueprintCircuit, ABC):
+    """Base class for linear system matrices."""
+
+    def __init__(self, num_state_qubits: int, tolerance: float, evo_time: float,
+                 name: str = 'ls_matrix') -> None:
+        """
+        Args:
+            num_state_qubits: the number of qubits where the unitary acts.
+            tolerance: the accuracy desired for the approximation
+            evo_time: the time of the Hamiltonian simulation
```

Comment:
Doe either of these make sense to have some usable/common default?

Reply:
changed to `evolution_time`